lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -95,6 +95,161 @@ def self.persist_buffers!
       true #TODO: More useful return value?
     end
 
+    # Check conflicts for Configuration Type Role Updates
+    def self.conflicts_for_operables(operable_ids, crs_ids)
+      descendants_query = <<-SQL
+        WITH RECURSIVE descendants AS (
+          SELECT parent_id, child_id, parent_id as root_node FROM assignments
+          WHERE parent_id IN (#{operable_ids.join(", ")})
+          UNION ALL
+          SELECT a.parent_id, a.child_id, root_node FROM assignments a
+          INNER JOIN descendants d 
+            ON a.parent_id = d.child_id
+            WHERE a.parent_id = d.child_id
+        )
+        SELECT root_node, ra.operator_uri, crs.id FROM descendants d
+        JOIN policy_element_associations AS pea
+          ON pea.object_attribute_id = d.child_id OR pea.object_attribute_id = d.parent_id
+        JOIN policy_elements as ra
+          ON ra.id = pea.user_attribute_id
+        JOIN policy_elements as os
+          ON pea.operation_set_id = os.id
+        JOIN assignments as bb_a
+          ON os.id = bb_a.parent_id
+        JOIN assignments as crs_a
+          ON bb_a.child_id = crs_a.child_id
+        JOIN policy_elements AS crs
+          ON crs_a.parent_id = crs.id
+        AND crs.active_policy_element_type = 'ConflictingRolesSet'
```

Comment:
This is Dalton-specific logic. It doesn't belong in the policy machine.

Reply:
active_policy_element is indeed Dalton-specific logic. However, pulling this into Dalton would make the conflicting roles check active record adapter-specific logic. There are some downsides to both approaches, so I suggest we move forward as is and have a design meeting when @CGDS gets back about whether or not we should extend the policy machine with a new conflicting roles set policy element.